torch_frame/data/multi_nested_tensor.py

```diff
@@ -0,0 +1,197 @@
+        values (Tensor): The values Tensor.
+        offset (Tensor): The offset Tensor.
+
+
+    """
+    def __init__(self, num_rows: int, num_cols: int, values: Dict[str, Tensor],
+                 offset: Tensor):
+        assert offset[0] == 0
+        assert offset[-1] == len(values)
+        assert len(offset) == num_rows * num_cols + 1
+        self.num_rows = num_rows
+        self.num_cols = num_cols
+        self.values = values
+        self.offset = offset
+
+    @classmethod
+    def from_tensor_mat(
+        cls,
+        tensor_mat: List[List[Tensor]],
+    ) -> 'MultiNestedTensor':
+        r"""Construct :class:`MultiNestedTensor` object from
+        :obj:`tensor_mat`.
+
+        Args:
+            tensor_mat List[List[Tensor]]: A dictionary of
+                matrix of PyTorch Tensors. :obj:`tensor_mat[i][j]` contains
+                1-dim PyTorch Tensor of :obj:`i`-th row and :obj:`j`-th column.
+
+        Returns:
+            MultiNestedTensor: Returned class object.
+        """
+        num_rows = len(tensor_mat)
+        num_cols = len(tensor_mat[0])
+
+        offset_list = []
+        accum_idx = 0
+        offset_list.append(accum_idx)
+        values_list = []
+        for i in range(num_rows):
+            for j in range(num_cols):
+                tensor = tensor_mat[i][j]
+                if not isinstance(tensor, Tensor):
+                    raise RuntimeError(
+                        "The element of tensor_mat must be PyTorch Tensor")
+                values_list.append(tensor)
+                accum_idx += len(tensor)
+                offset_list.append(accum_idx)
+
+        values = torch.cat(values_list)
+        offset = torch.tensor(offset_list, dtype=torch.long)
+
+        return cls(num_rows, num_cols, values, offset)
+
+    def __repr__(self) -> str:
+        name = self.__class__.__name__
+        name += f"(num_rows: {self.num_rows}, num_cols: {self.num_cols})"
+        return name
+
+    def __copy__(self) -> 'MultiNestedTensor':
+        out = self.__class__.__new__(self.__class__)
+        for key, value in self.__dict__.items():
+            out.__dict__[key] = value
+
+        out.values = copy.copy(out.values)
+        out.offset = copy.copy(out.offset)
+
+        return out
+
+    def __getitem__(
+        self,
+        index: Any,
+    ) -> Union['MultiNestedTensor', Tensor]:
+        if isinstance(index, Tuple):
+            # Get an element of (i, j). Returns Tensor
+            assert len(index) == 2
+            if index[0] < 0 or index[0] >= self.num_rows:
+                raise IndexError(
+                    f"Index out-of-bounds. The first element of {index} needs "
+                    f"to be [0, {self.num_rows}]")
+            if index[1] < 0 or index[1] >= self.num_cols:
+                raise IndexError(
+                    f"Index out-of-bounds. The second element of {index} "
+                    f"needs to be [0, {self.num_cols}]")
```

Comment:
Why would that raise an error? We should support negative indices.

Reply:
Good point.